crates/wasmparser/src/resources.rs

```diff
@@ -50,6 +50,9 @@ pub trait WasmModuleResources {
     /// The sub type must be canonicalized.
     fn sub_type_at(&self, type_index: u32) -> Option<&SubType>;
 
+    /// Returns the `SubType` associated with the given core type id.
+    fn sub_type_at_id(&self, id: CoreTypeId) -> Option<&SubType>;
```

Comment:
Unlike the other method in this trait which are called with possibly-arbitrary indices we know that `CoreTypeId` is always trusted so it's ok to return `&SubType` from this instead of `Option<_>`